source/IOSResolver/src/IOSResolver.cs

```diff
@@ -1843,7 +1862,7 @@ public enum CocoapodsIntegrationMethod {
                 }
                 AddPodInternal(
                     podName,
-                    "Unity-iPhone",
+                    matchedTarget ?? "Unity-iPhone",
```

Comment:
I'm a little confused by this, matchedTarget isn't present then something has gone wrong with the parsing.  Perhaps warn and don't add the pod to the wrong target instead?